cmd/builder/internal/version.go

```diff
@@ -5,22 +5,37 @@ package internal // import "go.opentelemetry.io/collector/cmd/builder/internal"
 
 import (
 	"fmt"
+	"log"
+	"runtime/debug"
 
 	"github.com/spf13/cobra"
 )
 
 var (
-	version = "dev"
+	version = ""
 	date    = "unknown"
 )
 
+// binVersion returns the version of the binary.
+// If the version is not set, it attempts to read the build information.
+func binVersion() string {
+	if version != "" {
+		return version
+	}
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		log.Fatal("Failed to read build info")
```

Comment:
Return an error instead of crashing.

Reply:
Changed to return errro.

https://github.com/open-telemetry/opentelemetry-collector/pull/8994/files#diff-c3eff1de420450a694b19343e4543ca062f42962c968262c00d776504810861cR21-R29